client/sites/hooks/use-transfer-with-software-status-query.ts

```diff
@@ -0,0 +1,37 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+type TransferWithSoftwareStatusResponse = {
+	software_transfer_status: string;
+};
```

Comment:
Given that we're querying a back-end API where the payload may change in future, it feels like it may be worth declaring this as an `interface` rather than a `type`.

Reply:
Also, do we want to distinguish between the transfer status and the software status? It feels like we may want two values to be returned so the UI can show progress bars etc more effectively.